tests/fixtures/git_repo.py

```diff
@@ -926,11 +926,11 @@ def simulate_change_commits_n_rtn_changelog_entry(
     file_in_repo: str,
 ) -> SimulateChangeCommitsNReturnChangelogEntryFn:
     def _simulate_change_commits_n_rtn_changelog_entry(
-        git_repo: Repo, commit_msgs: Sequence[CommitDef]
+        git_repo: Repo, commit_msgs: Sequence[CommitDef], file_path: str | None = None
```

Comment:
This is the change described in the PR title. If I create commits in the feature branch before merging the default branch, it leads to merge conflicts. Making changes in a different file avoids this issue.

Reply:
This is actually not necessary.  The `file_in_repo` file's contents actually doesn't matter at all for the use of PSR. We just need a change in a tracked file in order to create commits.  We really should just use the `--strategy-option` of git merge to forcefully resolve conflicts (the result of the file does not matter).  

I just checked the `build_repo_from_definition` fixture and unfortunately I did not implement the passing of a strategy option for a GIT_MERGE but I did for GIT_SQUASH. We should just carry that implementation over and then follow what I did in the `tests/fixtures/repos/github_flow/repo_w_default_release.py` to pass a `strategy_option` value in the repo definition.

Reply:
It's a very good point. Added `strategy_option` in this commit and removed `file_path` https://github.com/python-semantic-release/python-semantic-release/pull/1268/commits/1b8cf1eba7ef3b2ac0bf2c2696a83946344c493c.

Made it `NotRequired` to not update existing places where merge action is used (but I can do it if you think it would be better).  
And if you see something that is easy for you to fix and it's blocking a merge - I'm definitely not against you making changes in this PR to move it forward.